Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -258,18 +261,24 @@ void IOWindow::CreateMainLayout()
 
   m_devices_combo = new QComboBox();
   m_option_list = new QTableWidget();
-  m_select_button = new QPushButton(tr("Select"));
-  m_detect_button = new QPushButton(tr("Detect"), this);
-  m_test_button = new QPushButton(tr("Test"), this);
+
+  m_select_button =
+      new QPushButton(m_type == IOWindow::Type::Input ? tr("Insert Input") : tr("Insert Output"));
+  m_detect_button = new QPushButton(tr("Detect Input"), this);
+  m_test_button = new QPushButton(tr("Test Output"), this);
   m_button_box = new QDialogButtonBox();
   m_clear_button = new QPushButton(tr("Clear"));
   m_scalar_spinbox = new QSpinBox();
 
-  m_parse_text = new InputStateLineEdit([this] {
-    const auto lock = m_controller->GetStateLock();
-    return m_reference->GetState<ControlState>();
-  });
-  m_parse_text->setReadOnly(true);
+  if (m_type == Type::Input)
+  {
+    m_parse_text = new InputStateLineEdit([this] { return m_reference->GetState<ControlState>(); });
```

Comment:
Why is the lock no longer needed here?

Reply:
Oops, looks like it's still needed. I was thinking this is invoked from the `Update` event that already holds the lock, but that's not always the case.